src/wasm/wasm-binary.cpp

```diff
@@ -1242,6 +1239,44 @@ void WasmBinaryWriter::writeSourceMapProlog() {
     }
   }
 
+  // Remove unused function names from 'names' field.
+  if (!wasm->debugInfoSymbolNames.empty()) {
+    std::vector<std::string> newSymbolNames;
+    std::unordered_map<Index, Index> oldToNewIndex;
+
+    // Collect all used symbol name indexes.
+    for (auto& func : wasm->functions) {
+      for (auto& pair : func->debugLocations) {
```

Comment:
could you even do something like `[_, location]` here instead of having to use `pair.second`?